Mesh: Add lattice free-form deformation (#1814)
NVIDIA/physicsnemo: docs/api/nn/functionals/geometry.rst

:meth:`~physicsnemo.mesh.mesh.Mesh.morph`, or
:meth:`~physicsnemo.mesh.domain_mesh.DomainMesh.morph`.

Lattice Free-Form Deformation
-----------------------------

.. autofunction:: physicsnemo.nn.functional.ffd_points

.. code:: python

    import torch
    from physicsnemo.nn.functional import ffd_points

    points = torch.rand(1024, 3)
    control_displacements = torch.zeros(4, 4, 4, 3, requires_grad=True)

    deformed = ffd_points(
        points,
        control_displacements,
        origin=[0.0, 0.0, 0.0],
        extent=[1.0, 1.0, 1.0],
        basis="bernstein",
    )
    deformed.square().mean().backward()

> With zero control displacements the operation is exactly the identity, so a

Reviewer: ```suggestion
With zero control displacements, the operation is exactly the identity, so a
```

lattice initialized at zero is a well-behaved starting point for shape

> optimization: an optimizer—or a model producing the lattice

Reviewer: ```suggestion
optimization. An optimizer, or a model that produces the lattice
```

> displacements—learns the deformation from a differentiable objective on

Reviewer: ```suggestion
displacements, learns the deformation from a differentiable objective on
```
